### ISSUE TITLE
[Google cloud script] Trouble with setup doubts

### ISSUE THREAD

**Issue by LucianaMarques** [mlbench/mlbench-helm#6]
Hi there,

I'm trying to setup the project with the setup script and it's happened this:
```

$ ./google_cloud_setup.sh create-cluster
WARNING: Starting in 1.12, new clusters will have basic authentication disabled by default. Basic authentication can be enabled (or disabled) manually using the `--[no-]enable-basic-auth` flag.
WARNING: Starting in 1.12, new clusters will not have a client certificate issued. You can manually enable (or disable) the issuance of the client certificate using the `--[no-]issue-client-certificate` flag.
WARNING: Currently VPC-native is not the default mode during cluster creation. In the future, this will become the default mode and can be disabled using `--no-enable-ip-alias` flag. Use `--[no-]enable-ip-alias` flag to suppress this warning.
WARNING: Starting in 1.12, default node pools in new clusters will have their legacy Compute Engine instance metadata endpoints disabled by default. To create a cluster with legacy instance metadata endpoints disabled in the default node pool, run `clusters create` with the flag `--metadata disable-legacy-endpoints=true`.
This will enable the autorepair feature for nodes. Please see https://cloud.google.com/kubernetes-engine/docs/node-auto-repair for more information on node autorepairs.
WARNING: The behavior of --scopes will change in a future gcloud release: service-control and service-management scopes will no longer be added to what is specified in --scopes. To use these scopes, add them explicitly to --scopes. To use the new behavior, set container/new_scopes_behavior property (gcloud config set container/new_scopes_behavior true).
WARNING: Starting in Kubernetes v1.10, new clusters will no longer get compute-rw and storage-ro scopes added to what is specified in --scopes (though the latter will remain included in the default --scopes). To use these scopes, add them explicitly to --scopes. To use the new behavior, set container/new_scopes_behavior property (gcloud config set container/new_scopes_behavior true).
ERROR: (gcloud.container.clusters.create) ResponseError: code=400, message=No valid versions with the prefix "1.10" found.
Fetching cluster endpoint and auth data.
ERROR: (gcloud.container.clusters.get-credentials) ResponseError: code=404, message=Not found: projects/mlbench-test/zones/europe-west1-b/clusters/rel-2.
No cluster named 'rel-2' in mlbench-test.
Error from server (AlreadyExists): serviceaccounts "tiller" already exists
Error from server (AlreadyExists): clusterrolebindings.rbac.authorization.k8s.io "tiller" already exists
Creating /home/luciana/.helm 
Creating /home/luciana/.helm/repository 
Creating /home/luciana/.helm/repository/cache 
Creating /home/luciana/.helm/repository/local 
Creating /home/luciana/.helm/plugins 
Creating /home/luciana/.helm/starters 
Creating /home/luciana/.helm/cache/archive 
Creating /home/luciana/.helm/repository/repositories.yaml 
Adding stable repo with URL: https://kubernetes-charts.storage.googleapis.com 
Adding local repo with URL: http://127.0.0.1:8879/charts 
$HELM_HOME has been configured at /home/luciana/.helm.

Tiller (the Helm server-side component) has been installed into your Kubernetes Cluster.

Please note: by default, Tiller is deployed with an insecure 'allow unauthenticated users' policy.
To prevent this, run `helm init` with the --tiller-tls-verify flag.
For more information on securing your installation see: https://docs.helm.sh/using_helm/#securing-your-helm-installation
Happy Helming!

```

"mlbench-test" is the project title I created. 
I'm not quite understanding what's going on here.
- The first error accuses "the prefix "1.10" found", what exactly does this prefix stand for?
- The second error tries to get a credential and fails. I think it tries to search for the europe-west1-b zone... using this zone is mandatory when creating a cluster for this project?
And I believe the following errors are consequent of the first two. 

Thanks!

**Comment by LucianaMarques** [comment]
Also, when I try to run the "get-credentials" command, it shows nothing and I have to force the process to finish with ^C... did I miss something?

**Comment by Panaetius** [comment]
That's weird, it looks like it complains that kubernetes version 1.10 is not available. Can you try it with explicitly setting the `CLUSTER_VERSION` to one of these:
https://i.imgur.com/XlJpBqc.png

It worked fine when I used the script on Tuesday without setting the version.

What OS are you on?

As for the second question, it's `get-credential` without the 's'. But all that does is to run the same function as the part of `create-cluster` that gives you the `ERROR: (gcloud.container.clusters.get-credentials) ResponseError: code=404, message=Not found: projects/mlbench-test/zones/europe-west1-b/clusters/rel-2.` error. So if it doesn't hang then, it shouldn't hang when you call that part specifically.

FYI, I just created #7 which you didn't specifically mention, but is an issue nonetheless.

**Comment by LucianaMarques** [comment]
> What OS are you on?

I'm using Ubutun 18.04, do you think the user's OS could be a problem?

> FYI, I just created #7 which you didn't specifically mention, but is an issue nonetheless.
Nice! I can try to help later.

As soon as I get home I'll make the test with the cluster version you asked!

**Comment by Panaetius** [comment]
I don't think the OS is an issue, it's just important to ask in case someone uses something exotic like Cygwin on Windows.

Can you also tell me your gcloud version with `gcloud --version` ? It works for me on `Google Cloud SDK 234.0.0`

**Comment by LucianaMarques** [comment]
> I don't think the OS is an issue, it's just important to ask in case someone uses something exotic like Cygwin on Windows.

Haha indeed. I understand that now the project runs on linux (and perhaps mac), but did you discuss previously on making it available for Windows users?

> Can you also tell me your gcloud version with `gcloud --version` ? It works for me on `Google Cloud SDK 234.0.0`

```
$ gcloud --version
Google Cloud SDK 234.0.0
bq 2.0.41
core 2019.02.08
gsutil 4.36
Updates are available for some Cloud SDK components.  To install them,
please run:
  $ gcloud components update
```

I'm installing the updates later, but just wanted to share the original version in which I was trying to test yesterday.

**Comment by Panaetius** [comment]
No we haven't discussed making it available to windows users. But adding a bat file alongside the bash file shouldn't be too hard. Definitely not a priority right now, though.

Your gcloud versions look exactly the same as mine, except for some additional components installed on my end. But those shouldn't influence cluster setup at all. 

I'll wait on your feedback regarding specifying a full kubernetes version when creating the cluster.

**Comment by Panaetius** [comment]
I just ran into the same issue, it looks like when I was testing on Tuesday, I had an earlier version of gcloud, which has been updated since and leads to this problem. As per https://cloud.google.com/kubernetes-engine/release-notes#version-updates Kubernetes version 1.10 is no longer available. 

Working on fixing it now

**Comment by Panaetius** [comment]
Closed by e5af11dc875862dc73ab3af64954439bd0419f2d

**Comment by LucianaMarques** [comment]
I just successfully got an mlbench instance running in my browser now, thanks!
